### PR TITLE
[Utils] Install and Uninstall Script Improvements

### PR DESCRIPTION
**Pull request by SAtacker**
* Suggestion menstioned in https://github.com/WasmEdge/WasmEdge/issues/514#issue-1031461951
* Uses bash variable expansion to make libsome.so.xx.yy.zz to libsome.so and libsome.so.xx
* Minor: Removed exit trap for help

Signed-off-by: Shreyas Atre <shreyasatre16@gmail.com>

### CODE REVIEW

**Comment by SAtacker** [comment]
> For the above changes, I think the uninstaller should be updated, too?

I'll update it as well. Thanks.

**Comment by SAtacker** [comment]
> > If we can run a script in GitHub CI, I can make a detailed installation testing script for Mac and Linux.
> 
> Sure thing, please update the `test-install-script` to make a detailed testing of our install script. Currently, it supports both Mac(Intel model) and Linux(CentOS & Ubuntu)

Hi, instead of making a completely new script I have added a few checks for the recent modifications done. Let me know if any more of it or similar is required. Thanks.

**Review by q82419 — APPROVED** [review]

File: utils/install.sh [hunk 1]
```diff
@@ -396,7 +399,16 @@ install_wasmedge_tensorflow() {
 
 install_image_extensions() {
     if [ $IM_EXT_COMPAT == 1 ]; then
-        get_wasmedge_image_deps
+
+        [ "$EXT_V_SET_WASMEDGE_IM" -eq 0 ] && VERSION_IM=$VERSION &&
```
[the remark this answers is not in the document]
Comment (SAtacker) [reply utils/install.sh:429]: My output:
```
root@dad6fde7f05a:/# wget -qO- https://raw.githubusercontent.com/SAtacker/WasmEdge/fix_install_script/utils/install.sh | bash -s -- -e image
sudo could not be found
Detected Linux-x86_64
sudo could not be found
Installation path found at /root/.wasmedge
Removing /root/.wasmedge/env
Removing /root/.wasmedge/lib
Removing /root/.wasmedge/bin
Removing /root/.wasmedge/include
Removing /root/.wasmedge
No path provided
Installing in /root/.wasmedge
WasmEdge Installation at /root/.wasmedge
Fetching WasmEdge-0.8.2
WasmEdge-0.8.2-manylinux2014_x86_6 100%[================================================================>]  39.00M  3.55MB/s    in 11s     
Installing WasmEdge-0.8.2-Linux in /root/.wasmedge/include 
Installing WasmEdge-0.8.2-Linux in /root/.wasmedge/lib 
Installing WasmEdge-0.8.2-Linux in /root/.wasmedge/bin 
Installation of wasmedge-0.8.2 successfull
Installation of wasmedgec-0.8.2 successfull
Image Extensions
Fetching WasmEdge-image-deps-0.8.2
WasmEdge-image-deps-0.8.2-manylinu 100%[================================================================>] 715.18K  3.74MB/s    in 0.2s    
Fetching WasmEdge-image-0.8.2
WasmEdge-image-0.8.2-manylinux2014 100%[================================================================>] 637.30K  --.-KB/s    in 0.09s   

source /root/.wasmedge/env to use wasmedge binaries
```

File: utils/install.sh [hunk 2]
```diff
@@ -396,7 +425,16 @@ install_wasmedge_tensorflow() {
...
+            remote_version_availabilty second-state/WasmEdge-image "$VERSION_IM"
+
+        [ "$EXT_V_SET_WASMEDGE_IM_DEPS" -eq 0 ] && VERSION_IM_DEPS=$VERSION
+
+        [ "$(printf %s\\n%s\\n "$VERSION_IM_DEPS" "0.8.2")" == "$(printf %s\\n%s "$VERSION_IM_DEPS" "0.8.2" | sort --version-sort)" ] &&
+            remote_version_availabilty second-state/WasmEdge-image "$VERSION_IM_DEPS" &&
+            get_wasmedge_image_deps
+
```
Comment (SAtacker) [inline utils/install.sh:437]: @q82419 This part checks if the selected version is greater than 0.8.2 then this will not get executed further.

Comment (SAtacker) [reply utils/install.sh:437]: Tests:
```
root@3fb60088d61a:/# VERSION_IM_DEPS=0.8.1
root@3fb60088d61a:/#  [ "$(printf %s\\n%s\\n "$VERSION_IM_DEPS" "0.8.2")" == "$(printf %s\\n%s "$VERSION_IM_DEPS" "0.8.2" | sort --version-sort)" ] && echo "install" || echo "don't"
install
root@3fb60088d61a:/# VERSION_IM_DEPS=0.8.2
root@3fb60088d61a:/#  [ "$(printf %s\\n%s\\n "$VERSION_IM_DEPS" "0.8.2")" == "$(printf %s\\n%s "$VERSION_IM_DEPS" "0.8.2" | sort --version-sort)" ] && echo "install" || echo "don't"
install
root@3fb60088d61a:/# VERSION_IM_DEPS=0.9.0-rc.1
root@3fb60088d61a:/#  [ "$(printf %s\\n%s\\n "$VERSION_IM_DEPS" "0.8.2")" == "$(printf %s\\n%s "$VERSION_IM_DEPS" "0.8.2" | sort --version-sort)" ] && echo "install" || echo "don't"
don't
root@3fb60088d61a:/#
```

Comment (q82419) [reply utils/install.sh:437]: Thanks. I tested that and it really did a great job.

**Review by q82419 — APPROVED** [review]

**Pull request merged**